test/cql-pytest/test_uda.py

```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
+# Copyright 2020-present ScyllaDB
+#
+# This file is part of Scylla.
+#
+# Scylla is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Scylla is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
+
+#############################################################################
+# Tests for user-defined aggregates (UDA)
+#############################################################################
+
+from util import unique_name, new_test_table, new_function, new_aggregate
+
+# Test that computing an average by hand works the same as
+# the built-in function
+def test_custom_avg(scylla_only, cql, test_keyspace):
+    schema = 'id bigint primary key'
+    with new_test_table(cql, test_keyspace, schema) as table:
+        for i in range(8):
+            cql.execute(f"INSERT INTO {table} (id) VALUES ({10**i})")
+        avg_partial_body = "(state tuple<bigint, bigint>, val bigint) CALLED ON NULL INPUT RETURNS tuple<bigint, bigint> LANGUAGE lua AS 'return {state[1] + val, state[2] + 1}'"
+        div_body = "(state tuple<bigint, bigint>) CALLED ON NULL INPUT RETURNS bigint LANGUAGE lua AS 'return state[1]//state[2]'"
+        with new_function(cql, test_keyspace, avg_partial_body) as avg_partial, new_function(cql, test_keyspace, div_body) as div_fun:
+            custom_avg_body = f"(bigint) SFUNC {avg_partial} STYPE tuple<bigint, bigint> FINALFUNC {div_fun} INITCOND (0,0)"
+            with new_aggregate(cql, test_keyspace, custom_avg_body) as custom_avg:
+                custom_res = [row for row in cql.execute(f"SELECT {test_keyspace}.{custom_avg}(id) AS result FROM {table}")]
+                avg_res = [row for row in cql.execute(f"SELECT avg(id) AS result FROM {table}")]
+                assert custom_res == avg_res
```

Comment:
Please add tests that you can't create an aggregate if scalar components are missing, can't drop scalars if they're used, and that you can't use the aggregate after it's dropped. Also mismatches of argument types etc.